pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -968,6 +970,67 @@ func (qjm *XController) nodeChecks(histograms map[string]*dto.Metric, aw *arbv1.
 	return ok
 }
 
+// Function for calculating required resources to run AppWrapper and printing a log if there is a required increase for resources.
+func calculateRequiredResources(namespace string, appwrapperName string, requestedResources string, totalAvailable string) {
+	cpuRegex := regexp.MustCompile(`cpu (\d+(\.\d+)?)`)
+	memoryRegex := regexp.MustCompile(`memory (\d+(\.\d+)?)`)
+	gpuRegex := regexp.MustCompile(`GPU (\d+)`)
```

Comment:
Perhaps these could be global variables and save time by avoiding compiling the same regex pattern on each call?
One more advantage is that in case of error, `MustCompile()` will panic immediately at start time rather than at the first call of `calculateRequiredResources()`

Reply:
Good idea I'll make changes